src/extensions/averagewaveforms/AverageWaveformsView/fetch_average_waveforms_2.py

```diff
@@ -38,6 +39,12 @@ def fetch_average_waveform_2(snippets_h5, unit_id):
         unit_spike_train = np.array(f.get(f'unit_spike_trains/{unit_id}'))
         unit_waveforms = np.array(f.get(f'unit_waveforms/{unit_id}/waveforms'))
         unit_waveforms_channel_ids = np.array(f.get(f'unit_waveforms/{unit_id}/channel_ids'))
+
+        if visible_channel_ids is not None:
+            inds = [ii for ii in range(len(unit_waveforms_channel_ids)) if unit_waveforms_channel_ids[ii] in visible_channel_ids]
+            unit_waveforms_channel_ids = unit_waveforms_channel_ids[inds]
+            unit_waveforms = unit_waveforms[:, inds, :]
```

Comment:
Ah, this is a numpy higher-dimension slice, right? So it says "from all elements in unit_waveforms, return the elements corresponding to the indices of the visible unit waveforms' channel ids, and return all elements within those", or something like that?

Reply:
yes